SchoolApplication/School.Repository/StudentRepository.cs

```diff
@@ -14,123 +14,177 @@ public class StudentRepository : BaseRepository<Student>
                     {
-                        StudentCourse stdCourse = new StudentCourse();
-                    // var Std = db.Students.OrderByDescending(x => x.Id).FirstOrDefault();
-                        db.Students.Add(obj);
-                        stdCourse.StudentId = obj.Id;
-                        var courseList = courses.Split(',').ToList();
-                        foreach (var course in courseList)
+                        foreach (var Course in stdPost.Courses)
                         {
-                            stdCourse.CourseId = Convert.ToInt32(course);
-                            db.StudentCourses.Add(stdCourse);
-                            db.SaveChanges();
+                            StudentCourse course_Obj = new StudentCourse();
+                            course_Obj.StudentId = stdPost.student.Id;
+                            course_Obj.CourseId = Convert.ToInt32(Course);
+                            courseList.Add(course_Obj);
+
                         }
-                   
+                        db.StudentCourses.AddRange(courseList);
+                        db.SaveChanges();
+
                     }
-                }
-                catch (Exception e)
-                {
 
-                    return false;
+
                 }
+                //Adding updated  Courses
+
+            
+            }
+            catch (Exception e)
+            {
+                return false;
+            }
 
             return true;
 
         }
-        //public bool DeleteStudent(int id)
-        //{
-        //    try
-        //    {
-        //        using (var context = new SchoolContext())
-        //        {
-        //            context.Students.Remove(context.Students.Where(x => x.Id == id).FirstOrDefault());
-        //            context.SaveChanges();
-        //            return true;
-        //        }
-        //    }
-        //    catch (Exception)
-        //    {
-
-        //        return false;
-        //    }
-
-
-        //}
-        public Student FindStudent(int id)
+        public bool DeleteStudent(int id)
         {
             try
             {
                 using (var context = new SchoolContext())
                 {
-                    var student = context.Students.Where(x => x.Id == id).FirstOrDefault();
-                    return student;
+                   var std= context.Students.Where(x => x.Id == id).FirstOrDefault();
+                    //  StudentCourse stdCourse = new StudentCourse();
+                    // stdCourse.StudentId = std.Id;
+                    var stdcourses = context.StudentCourses.Where(x => x.StudentId == id);
+                    foreach (var course in stdcourses)
+                    {
+                        context.StudentCourses.Remove(course);
+                        context.SaveChanges();
+                    }
+                    context.Students.Remove(std);
+                    context.SaveChanges();
```

Comment:
saving changes inside the loop and again after the loop ends . Need to save changes once after the loop ends.